src/Illuminate/Queue/Worker.php

```diff
@@ -597,7 +647,27 @@ protected function calculateBackoff($job, WorkerOptions $options)
                         : $options->backoff
         );
 
-        return (int) ($backoff[$job->attempts() - 1] ?? last($backoff));
+        $backoffMode = $job->backoffMode();
+
+        if ($backoffMode === Job::BACKOFF_FROM_ATTEMPTS) {
```

Comment:
Since this is the default case, I'd think you should save it for later.
First check if the backoff mode is calculated by exceptions and return its backoff calculation. Then return this default calculation mode later on.
```suggestion
        if ($backoffMode === Job:: BACKOFF_FROM_EXCEPTIONS) {
            // back off from exceptions calculation
         }
        
        // default calculation of backoff
         return (int) ($backoff[$job->attempts() - 1] ?? last($backoff));
```

Reply:
Thanks, I was thinking about the default calculation mode also. But then I think maybe it’s better to throw an exception  when the backoffMode’s value is unsupported.
What do you think?

Reply:
I think it's just safer to use the default backoff mode.

Reply:
ok!